### ISSUE TITLE
5.4节关于TensorFlow的描述有误

### ISSUE THREAD

**Issue by basicv8vc** [nndl/textbook-comments#7]
5.4节自动梯度计算 中指出“Theano和TensorFlow都采用了符号微分的方法进行自动求解梯度。”

实际上，TensorFlow中计算梯度采用的是自动微分(automatic differentiation)，在TensorFlow的Tutorials中有明确描述：

* Gradients (AKA automatic differentiation): Given a graph and a list of input and output operations, add operations to the graph that compute the partial deriviatives (gradients) of the inputs with respect to the outputs. Allows for customization of the gradient function for a particular operation in the graph. https://www.tensorflow.org/versions/r0.11/how_tos/language_bindings/index.html

* Now that we have defined our model and training loss function, it is straightforward to train using TensorFlow. Because TensorFlow knows the entire computation graph, it can use automatic differentiation to find the gradients of the loss with respect to each of the variables. https://www.tensorflow.org/versions/r0.11/tutorials/mnist/pros/index.html

* Given a graph of ops, TensorFlow uses automatic differentiation (backpropagation) to add new ops representing gradients with respect to the existing ops. https://www.tensorflow.org/versions/r0.12/how_tos/adding_an_op/index.html

**Comment by xpqiu** [comment]
多谢！更多的资料显示是符号微分。

Because derivatives are “automatic”, some call it “autodiff” or “automatic differentiation”
This is NOT the traditional meaning of automatic differentiation (AD) (Griewank & Walther, 2008)

http://www.cs.nuim.ie/~gunes/files/Baydin-MSR-Slides-20160201.pdf
